plugins/renamerOnUpdate/renamerOnUpdate.py

```diff
@@ -347,163 +361,319 @@ def exit_plugin(msg=None, err=None):
+    # by Parent
+    if STASH_SCENE["studio"].get("parent_studio"):
+        if config.studio_templates.get(STASH_SCENE["studio"]["name"]):
+            filename_template = config.studio_templates[STASH_SCENE["studio"]["name"]]
+
+# Change by Tag
+if STASH_SCENE.get("tags") and config.tag_templates:
+    for tag in STASH_SCENE["tags"]:
+        if config.tag_templates.get(tag["name"]):
+            filename_template = config.tag_templates[tag["name"]]
             break
 
-# IF YOU WANT TO FORCE A TEMPLATE FOR ANY FILE, UNCOMMENT BELOW
-#result_template = "$date $title"
+#                           END                                    #
+####################################################################
 
-# END OF PERSONAL THINGS
+if not filename_template:
+    exit_plugin("No template for this scene.")
 
-if result_template is None:
-    exit_plugin("No template for this file.")
-else:
-    log.LogDebug("Using this template: {}".format(result_template))
+#log.LogDebug("Using this template: {}".format(filename_template))
 
-current_path = scene_info["path"]
+current_path = STASH_SCENE["path"]
+# note: contain the dot (.mp4)
 file_extension = os.path.splitext(current_path)[1]
-# Note: basename contains the extension
+# note: basename contains the extension
 current_filename = os.path.basename(current_path)
 current_directory = os.path.dirname(current_path)
 
 # Grabbing things from Stash
 scene_information = {}
+
 # Grab Title (without extension if present)
-if scene_info.get("title"):
-    # Remove extension
-    scene_information["title"] = re.sub(file_extension + '$', '', scene_info["title"])
+if STASH_SCENE.get("title"):
+    # Removing extension if present in title
+    scene_information["title"] = re.sub("{}$".format(file_extension), "", STASH_SCENE["title"])
+
 # Grab Date
-scene_information["date"] = scene_info.get("date")
-# Grab Performer (Dani Daniels Riley Reid)
-if scene_info.get("performers"):
+scene_information["date"] = STASH_SCENE.get("date")
+
+# Grab Performer
+if STASH_SCENE.get("performers"):
     perf_list = ""
-    if len(scene_info["performers"]) > 3:
-        log.LogWarning("More than 3 performers.")
+    if len(STASH_SCENE["performers"]) > PERFORMER_LIMIT:
+        log.LogInfo("More than {} performer(s). Ignoring $performer".format(PERFORMER_LIMIT))
     else:
-        for performer in scene_info["performers"]:
-            if performer.get("name"):
-                perf_list += performer["name"] + " "
+        for perf in STASH_SCENE["performers"]:
+            #log.LogDebug(performer)
+            if PERFORMER_IGNORE_MALE:
+                if perf["gender"] != "MALE":
+                    perf_list += perf["name"] + PERFORMER_SPLITCHAR
             else:
-                continue
-        perf_list = perf_list.strip()
+                perf_list += perf["name"] + PERFORMER_SPLITCHAR
+        # Remove last character
+        perf_list = perf_list[:-1]
```

Comment:
Might be better to use `perf_list = perf_list[:-len(PERFORMER_SPLITCHAR)]` instead, so this will remove longer splitchars as well.

Reply:
Good catch. Thanks you :)